### PR TITLE
Feat/share callout

### PR DESCRIPTION
**Pull request by rlafuente**
Now implemented with proper sharing links (only email missing, but posting the PR for review of the rest)

### CODE REVIEW

File: src/modules/callouts/components/CalloutSharingPanel.vue [hunk 1]
```diff
@@ -0,0 +1,112 @@
+<template>
+  <div class="h-auto flex flex-col border border-primary-40 p-5 my-2">
+    Share this callout with the following address:
```
Comment (wpf500) [inline src/modules/callouts/components/CalloutSharingPanel.vue]: Needs to be localised

File: src/modules/callouts/components/CalloutSharingPanel.vue [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        mt-2
+        mb-4
+        pl-3
+      "
+    >
+      <span class="text-link mr-2">{{ calloutUrl }}</span>
+      <AppButton icon="copy" size="sm" @click="copyToClipboard">Copy</AppButton>
+    </div>
+
+    Or quickly share using one of these services:
```
Comment (wpf500) [inline src/modules/callouts/components/CalloutSharingPanel.vue]: Needs to be localised

File: src/modules/callouts/components/CalloutSharingPanel.vue [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        </a>
+      </div>
+      <div>
+        <a
+          :href="'mailto:?body=' + encodeURIComponent(calloutUrl)"
+          target="_blank"
+          rel="nofollow noopener"
+        >
+          <font-awesome-icon :icon="['fa', 'envelope']" class="mr-1" />
+          E-mail
```
Comment (wpf500) [inline src/modules/callouts/components/CalloutSharingPanel.vue]: Needs to be localised

File: src/modules/callouts/components/CalloutSharingPanel.vue [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+<script lang="ts" setup>
+import AppButton from '../../../components/forms/AppButton.vue';
+
+const props = defineProps<{
+  slug: string;
+}>();
+
+const baseUrl = import.meta.env.VITE_APP_BASE_URL;
+const calloutUrl = `${baseUrl}/${props.slug}`;
```
Comment (wpf500) [inline src/modules/callouts/components/CalloutSharingPanel.vue]: I think this should be:
```ts
const calloutUrl = `${baseUrl}/callouts/${props.slug}`
```

**Review by wpf500 — CHANGES_REQUESTED** [review]
Nice one! A few small changes requested but looks great :)

**Comment by rlafuente** [comment]
Updated!

**Review by wpf500 — APPROVED** [review]

**Pull request merged**